lib/net/ldap/connection.rb

```diff
@@ -51,6 +51,15 @@ def open_connection(server)
     hosts.each do |host, port|
       begin
         prepare_socket(server.merge(socket: @socket_class.new(host, port, socket_opts)))
+        if encryption
+          if encryption[:tls_options] &&
+             encryption[:tls_options][:verify_mode] &&
+             encryption[:tls_options][:verify_mode] == OpenSSL::SSL::VERIFY_NONE
+            warn "not verifying SSL hostname of LDAPS server"
+          else
+            @conn.post_connection_check(host)
+          end
+        end
```

Comment:
@JPvRiel what was your reasoning for putting this check in `#open_connection` rather than `.wrap_with_ssl`? It looks like that code path is always used when setting up SSL connections.